async-opcua-types/src/xml/encoding.rs

```diff
@@ -55,6 +55,12 @@ pub trait XmlEncodable: XmlType {
         writer: &mut XmlStreamWriter<&mut dyn Write>,
         context: &Context<'_>,
     ) -> EncodingResult<()>;
+
+    /// This method should return `true` if the value is default
+    /// and should not be serialized.
+    fn is_null_xml(&self) -> bool {
+        false
```

Comment:
Guess this method will never return `true`. Is that expected?

Reply:
This is the trait default, it's overridden in certain types.

Reply:
As it's defined in a trait, wouldn't it be better to not implement it as a default `false` but require it to be implemented by anyone that implements this trait for a type?

Reply:
We _could_ do that, I guess. It would be a lot of noise though, and most implementations would simply be `false`.

If we do this, I'd rather do it separately, since we would want to do the same for JSON. Maybe combine it with moving this to a separate trait?

Reply:
Just wondered if this could cause unexpected behavior were a type forgets to implement/overwrite the default. But as the default is `false` I guess that would be unlikely.